File: src/main/java/dev/lbuddyboy/flash/handler/RedisHandler.java
package dev.lbuddyboy.flash.handler;

import dev.lbuddyboy.flash.Flash;
import dev.lbuddyboy.flash.redis.sub.JedisSubscriber;
import dev.lbuddyboy.flash.util.bukkit.CC;
import lombok.Getter;
import org.bukkit.Bukkit;
import redis.clients.jedis.Jedis;
import redis.clients.jedis.JedisPool;
import redis.clients.jedis.JedisPoolConfig;

public class RedisHandler {

	@Getter private static boolean enabled;

	public RedisHandler() {
		if (!Flash.getInstance().getConfig().getBoolean("redis.use")) {
			Bukkit.getConsoleSender().sendMessage(CC.translate("&fDid not connect to the &bRedis Database&f &aenable&f it in the &econfig.yml&f!"));
			return;
		}
		connect();
	}

	public static void connect() {
		JedisPool connectTo = requestJedis();

		Thread thread = new Thread(() -> {
			while (true) {
				try {
					Jedis jedis = connectTo.getResource();
					Throwable throwable = null;
					try {
						JedisSubscriber pubSub = new JedisSubscriber();
						jedis.subscribe(pubSub, "Flash:Global");
					} catch (Throwable pubSub) {
						throwable = pubSub;
						throw pubSub;
					} finally {
						if (jedis == null) continue;
						if (throwable != null) {
							try {
								jedis.close();
							} catch (Throwable pubSub) {
								throwable.addSuppressed(pubSub);
							}
							continue;
						}
						jedis.close();
					}
				} catch (Exception e) {
					e.printStackTrace();
				}
			}
		}, "Redis Packet Subscriber");
		thread.setDaemon(true);
		thread.start();

		enabled = true;
	}

	public static JedisPool requestJedis() {
		return new JedisPool(new JedisPoolConfig(),
				Flash.getInstance().getConfig().getString("redis.host"), Flash.getInstance().getConfig().getInt("redis.port"), 20000,
				(Flash.getInstance().getConfig().getString("redis.auth.password").isEmpty() ? null : Flash.getInstance().getConfig().getString("redis.auth.password")),
				Flash.getInstance().getConfig().getInt("redis.channel-id", 0));
	}

}

File: src/main/java/dev/lbuddyboy/flash/handler/RedisHandler.kt
package dev.lbuddyboy.flash.handler

import dev.lbuddyboy.flash.Flash
import dev.lbuddyboy.flash.redis.sub.JedisSubscriber
import dev.lbuddyboy.flash.util.bukkit.CC
import lombok.Getter
import org.bukkit.Bukkit
import redis.clients.jedis.JedisPool
import redis.clients.jedis.JedisPoolConfig

class RedisHandler {
    init {
        if (!Flash.instance.config.getBoolean("redis.use")) {
            Bukkit.getConsoleSender()
                .sendMessage(CC.translate("&fDid not connect to the &bRedis Database&f &aenable&f it in the &econfig.yml&f!"))
            return
        }
        connect()
    }

    companion object {
        @Getter
        private var enabled = false
        fun connect() {
            val connectTo = requestJedis()
            val thread = Thread({
                while (true) {
                    try {
                        val jedis = connectTo.resource
                        var throwable: Throwable? = null
                        try {
                            val pubSub = JedisSubscriber()
                            jedis!!.subscribe(pubSub, "Flash:Global")
                        } catch (pubSub: Throwable) {
                            throwable = pubSub
                            throw pubSub
                        } finally {
                            if (jedis == null) continue
                            if (throwable != null) {
                                try {
                                    jedis.close()
                                } catch (pubSub: Throwable) {
                                    throwable.addSuppressed(pubSub)
                                }
                                continue
                            }
                            jedis.close()
                        }
                    } catch (e: Exception) {
                        e.printStackTrace()
                    }
                }
            }, "Redis Packet Subscriber")
            thread.isDaemon = true
            thread.start()
            enabled = true
        }

        @JvmStatic
		fun requestJedis(): JedisPool {
            return JedisPool(
                JedisPoolConfig(),
                Flash.instance.config.getString("redis.host"),
                Flash.instance.config.getInt("redis.port"),
                20000,
                if (Flash.instance.config.getString("redis.auth.password")
                        .isEmpty()
                ) null else Flash.instance.config.getString("redis.auth.password"),
                Flash.instance.config.getInt("redis.channel-id", 0)
            )
        }
    }
}